exercises/practice/forth/test/forth_test.dart

```diff
@@ -2,294 +2,391 @@ import 'package:forth/forth.dart';
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('1 -'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Stack empty')),
+    );
+  }, skip: true);
+}
+
+void multiplication() {
+  test('can multiply two numbers', () {
+    var forth = Forth();
+    forth.evaluate('2 4 *');
+    expect(forth.stack, equals(<int>[8]));
+  }, skip: true);
+
+  test('errors if there is nothing on the stack', () {
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('*'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Stack empty')),
+    );
+  }, skip: true);
+
+  test('errors if there is only one value on the stack', () {
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('1 *'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Stack empty')),
+    );
+  }, skip: true);
+}
+
+void division() {
+  test('can divide two numbers', () {
+    var forth = Forth();
+    forth.evaluate('12 3 /');
+    expect(forth.stack, equals(<int>[4]));
+  }, skip: true);
+
+  test('performs integer division', () {
+    var forth = Forth();
+    forth.evaluate('8 3 /');
+    expect(forth.stack, equals(<int>[2]));
+  }, skip: true);
+
+  test('errors if dividing by zero', () {
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('4 0 /'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Division by zero')),
+    );
+  }, skip: true);
+
+  test('errors if there is nothing on the stack', () {
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('/'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Stack empty')),
+    );
+  }, skip: true);
+
+  test('errors if there is only one value on the stack', () {
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('1 /'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Stack empty')),
+    );
+  }, skip: true);
+}
+
+void combinedArithmetic() {
+  test('addition and subtraction', () {
+    var forth = Forth();
+    forth.evaluate('1 2 + 4 -');
+    expect(forth.stack, equals(<int>[-1]));
+  }, skip: true);
+
+  test('multiplication and division', () {
+    var forth = Forth();
+    forth.evaluate('2 4 * 3 /');
+    expect(forth.stack, equals(<int>[2]));
+  }, skip: true);
+}
+
+void dup() {
+  test('copies a value on the stack', () {
+    var forth = Forth();
+    forth.evaluate('1 dup');
+    expect(forth.stack, equals(<int>[1, 1]));
+  }, skip: true);
+
+  test('copies the top value on the stack', () {
+    var forth = Forth();
+    forth.evaluate('1 2 dup');
+    expect(forth.stack, equals(<int>[1, 2, 2]));
+  }, skip: true);
+
+  test('errors if there is nothing on the stack', () {
+    var forth = Forth();
+    expect(
+      () => forth.evaluate('dup'),
+      throwsA(isA<Exception>().having((e) => e.toString(), 'message', 'Exception: Stack empty')),
```

Comment:
The error handling is a bit duplicative. Can we extract it into its own property and then reuse that property?

Same would go for any other error handling that is repeatedly used.

Reply:
Definitely!

Reply:
I've re-generated and pushed the branch.